temporalcli/commands.activity.go

```diff
@@ -70,3 +92,109 @@ func (c *TemporalActivityFailCommand) run(cctx *CommandContext, args []string) e
+		activityOptions.HeartbeatTimeout = durationpb.New(c.HeartbeatTimeout.Duration())
+		updatePath = append(updatePath, "heartbeat_timeout")
+	}
+
+	if c.Command.Flags().Changed("retry-initial-interval") ||
+		c.Command.Flags().Changed("retry-maximum-interval") ||
+		c.Command.Flags().Changed("retry-backoff-coefficient") ||
+		c.Command.Flags().Changed("retry-maximum-attempts") {
+		activityOptions.RetryPolicy = &common.RetryPolicy{}
+	}
+
+	if c.Command.Flags().Changed("retry-initial-interval") {
+		activityOptions.RetryPolicy.InitialInterval = durationpb.New(c.RetryInitialInterval.Duration())
+		updatePath = append(updatePath, "retry_policy.initial_interval")
+	}
+
+	if c.Command.Flags().Changed("retry-maximum-interval") {
+		activityOptions.RetryPolicy.MaximumInterval = durationpb.New(c.RetryMaximumInterval.Duration())
+		updatePath = append(updatePath, "retry_policy.maximum_interval")
+	}
+
+	if c.Command.Flags().Changed("retry-backoff-coefficient") {
+		activityOptions.RetryPolicy.BackoffCoefficient = float64(c.RetryBackoffCoefficient)
+		updatePath = append(updatePath, "retry_policy.backoff_coefficient")
+	}
+
+	if c.Command.Flags().Changed("retry-maximum-attempts") {
+		activityOptions.RetryPolicy.MaximumAttempts = int32(c.RetryMaximumAttempts)
+		updatePath = append(updatePath, "retry_policy.maximum_attempts")
+	}
+
+	result, err := cl.WorkflowService().UpdateActivityOptionsById(cctx, &workflowservice.UpdateActivityOptionsByIdRequest{
+		Namespace:       c.Parent.Namespace,
+		WorkflowId:      c.WorkflowId,
+		RunId:           c.RunId,
+		ActivityId:      c.ActivityId,
+		ActivityOptions: activityOptions,
+		UpdateMask: &fieldmaskpb.FieldMask{
+			Paths: updatePath,
+		},
+
+		Identity: c.Identity,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to update Activity options: %w", err)
+	}
+
+	updatedOptions := updateOptionsDescribe{
+		TaskQueue: result.GetActivityOptions().TaskQueue.GetName(),
+
+		ScheduleToCloseTimeout: toDuration(result.GetActivityOptions().ScheduleToCloseTimeout),
+		ScheduleToStartTimeout: toDuration(result.GetActivityOptions().ScheduleToStartTimeout),
+		StartToCloseTimeout:    toDuration(result.GetActivityOptions().StartToCloseTimeout),
+		HeartbeatTimeout:       toDuration(result.GetActivityOptions().HeartbeatTimeout),
+
+		InitialInterval:    toDuration(result.GetActivityOptions().RetryPolicy.InitialInterval),
+		BackoffCoefficient: result.GetActivityOptions().RetryPolicy.BackoffCoefficient,
+		MaximumInterval:    toDuration(result.GetActivityOptions().RetryPolicy.MaximumInterval),
+		MaximumAttempts:    result.GetActivityOptions().RetryPolicy.MaximumAttempts,
+	}
+
+	_ = cctx.Printer.PrintStructured(updatedOptions, printer.StructuredOptions{})
+
+	return nil
+}
+
+// Converts the duration to Go's native time.Duration.
+// Returns the zero time.Duration value for nil duration.
+func toDuration(duration *durationpb.Duration) (d time.Duration) {
+	if duration != nil {
+		d = duration.AsDuration()
```

Comment:
I think this call is already `nil` safe, not sure a new helper is needed

Reply:
yeah, looks both GetSeconds() and GetNanos() are nil sage.
Thus making  AsDuration nil safe as well.
I will remove it